src/main/java/bio/terra/service/resourcemanagement/MetadataDataAccessUtils.java

```diff
@@ -89,49 +127,75 @@ public AccessInfoModel accessInfoFromDataset(final Dataset dataset) {
       AzureStorageAccountResource storageAccountResource =
           resourceService.getDatasetStorageAccount(dataset, profileModel);
       return makeAccessInfoAzure(
-          dataset.getName(), storageAccountResource, dataset.getTables(), profileModel);
+          dataset, storageAccountResource, dataset.getTables(), profileModel, userRequest);
     } else {
       throw new IllegalArgumentException("Unrecognized cloud platform");
     }
   }
 
   private AccessInfoModel makeAccessInfoAzure(
-      final String datasetName,
+      final FSContainerInterface collection,
       final AzureStorageAccountResource storageAccountResource,
       final List<? extends Table> tables,
-      final BillingProfileModel profileModel) {
+      final BillingProfileModel profileModel,
+      final AuthenticatedUserRequest userRequest) {
     AccessInfoModel accessInfoModel = new AccessInfoModel();
 
+    BlobSasTokenOptions blobSasTokenOptions =
+        new BlobSasTokenOptions(
+            DEFAULT_SAS_TOKEN_EXPIRATION,
+            new BlobSasPermission().setReadPermission(true).setListPermission(true),
+            userRequest.getEmail());
+
+    String blobName;
+    Function2<FSContainerInterface, Table, String> tableBlobGenerator;
```

Comment:
`Function2` is a scala API. The Java API is `BiFunction`
```suggestion
    BiFunction <FSContainerInterface, Table, String> tableBlobGenerator;
```

Reply:
Oh derp...force of habit from my Spark days